docs/proposals/helm-charts.md

```diff
@@ -0,0 +1,169 @@
+    Tags:        chart.Keywords,
+    Description: chart.Description,
+    Metadata: map[string]interface{}{
+        //"longDescription":  chart.Description,
+        "displayName":      fmt.Sprintf("%s (Helm)", chart.Name),
+        "documentationUrl": chart.Home,
+        "dependencies":     chart.Sources,
+        "imageUrl":         chart.Icon,
+    },
+    Plans: []apb.Plan{
+        apb.Plan{
+            Name:        "default",
+            Description: "Default plan for running helm charts",
+            Parameters: []apb.ParameterDescriptor{
+                apb.ParameterDescriptor{
+                    Name:      "repo",
+                    Title:     "Helm Chart Repository URL",
+                    Type:      "string",
+                    Default:   r.Config.URL.String(),
+                    Pattern:   fmt.Sprintf("^%s$", r.Config.URL.String()),
+                    Updatable: false,
+                    Required:  false,
+                },
+                apb.ParameterDescriptor{
+                    Name:      "chart",
+                    Title:     "Helm Chart",
+                    Type:      "string",
+                    Default:   chart.Name,
+                    Pattern:   fmt.Sprintf("^%s$", chart.Name),
+                    Updatable: false,
+                    Required:  false,
+                },
+                apb.ParameterDescriptor{
+                    Name:      "name",
+                    Title:     "Release Name",
+                    Type:      "string",
+                    Default:   "helmrunner",
+                    Updatable: false,
+                    Required:  false,
+                },
+                apb.ParameterDescriptor{
+                    Name:        "values",
+                    Title:       "Values",
+                    Type:        "string",
+                    DisplayType: "textarea",
+                    Default:     values,
+                    Updatable:   false,
+                    Required:    false,
+                },
+            },
+        },
+    },
+}
+```
+
+Most of the data required to create a useful `Spec` object is contained in
+the Chart object. However, in order to retrieve the default values we will use
+[Helm's Chartutil
+pkg](https://github.com/kubernetes/helm/blob/master/pkg/chartutil/load.go) to
+load the archive specified at `Chart.URLs[0]` and read the values.
+
+## Phase 2 - Support Authenticated Chart Repositories
+
+Helm Chart Repositories can be authenticated, [this
+issue](https://github.com/kubernetes/helm/issues/1038) provides more
+information. The broker should support authenticated chart repositories.
+
+## Phase 3 - Chart Versions as Parameter
+
+In phase 1, only the latest chart version will be used. The broker should
+support all versions of a chart and allow the consumer to specify which version
+of a chart they wish to install. The base image would need to be updated to
+handle specifying a specific version at provision time.
```

Comment:
Where is the version information stored? What does a versioned chart url look like? Or more specifically how would someone get the specific version of a chart? What changes do you anticipate being done to the base image? Or is this another proposal for when we do phase 3?

Reply:
I went ahead and made this part of the initial proposal, it's easy to get all of the chart's versions.